x/evm/keeper/evm.go

```diff
@@ -73,6 +73,9 @@ func (k *Keeper) CallEVM(ctx sdk.Context, from common.Address, to *common.Addres
 		gp := k.GetGasPool()
 		value := utils.Big0
 		if val != nil {
+			if val.IsNegative() {
```

Comment:
we need this check for the `else` case as well so let's move it to before `if evm == nil`